### PR TITLE
Secured ros2 office demo

### PR DESCRIPTION
**Pull request by marcoag**
Adding the files and instructions needed for a secured run of the office demo.

### CODE REVIEW

File: README.md
Comment (Yadunund) [inline README.md:73]:
```suggestion
#### Secure ROS 2 Office Demo
```

File: README.md
Comment (Yadunund) [inline README.md:75]:
```suggestion
The office demo can be run in secure mode using the [ROS 2 DDS-Security integration](https://design.ros2.org/articles/ros2_dds_security.html) (SROS2) capabilities. This security features will provide encryption, authentication and access control to the whole ROS 2 setup of RMF.
```

File: README.md
Comment (Yadunund) [inline README.md:77]:
```suggestion
To use ROS 2 to secure the office demo a few environmental variables need to be set. Adding them to a script would make it easier to source them in any shell where we need to run ROS 2 secured nodes.
```

File: README.md
Comment (Yadunund) [inline README.md:83]: Should we add `export RMW_IMPLEMENTATION=rmw_cyclonedds_cpp` here and bring the Cyclone DDS installation instruction to above this instruction block?

File: README.md
Comment (Yadunund) [inline README.md:100]: Sourcing `~/rmf_demos_ws/install/setup.bash` will automatically source the ROS distro used to build the workspace as an underlay.

File: README.md
Comment (Yadunund) [reply README.md:83]: Also explicitly have instructions to clone the cyclone DDS repos into the workspace and checkout the `foxy` branch

**Review by Yadunund — CHANGES_REQUESTED** [review]
Still trying out the demo but some early comments

File: README.md
Comment (Yadunund) [inline README.md:155]:
```suggestion
gzclient --verbose ./rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world
```
```suggestion
gzclient --verbose ./install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world
```

File: README.md
Comment (Yadunund) [reply README.md:83]: Okay I understand now that we cannot have the`RMF_IMPLEMENTATION` line inside this file at this point because we are relying on FastDDS to generate the  security artifacts.

File: README.md
Comment (marcoag) [reply README.md:100]:
```suggestion
Because SROS2 does not support `ros2 launch` yet all the different nodes will have to be launched individually. A `yaml` file with the parameters of all the nodes is provided. One terminal per execution is recommended for an easier debug of the system. Make sure you `source ~/rmf_demos_ws/install/setup.bash` in each of them before executing the different commands.
```

**Review by Yadunund — APPROVED** [review]
Managed to successfully run the secured demo and it works great! Thanks for making the necessary changes. Could you update the changelogs before merging?

**Comment by marcoag** [comment]
@Yadunund could you test the `tmux` script?

https://github.com/osrf/rmf_demos/pull/135/commits/85f4bfb714222989d64ace34e3ab00e6f7fde0d3

File: demos/sros2/office_deploy.bash
Comment (Yadunund) [inline demos/sros2/office_deploy.bash:85]:
```suggestion
echo "export GAZEBO_MODEL_PATH=$WORKSPACE/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/models:$WORKSPACE/install/rmf_demo_assets/share/rmf_demo_assets/models:/usr/share/gazebo-11/models
```

**Review by Yadunund — CHANGES_REQUESTED** [review]
The tmux script works well with the proposed change. We can merge this in once the Rviz segfault is fixed

**Comment by Yadunund** [comment]
Also, what do you think of adding `rmw_cyclonedds` and `cyclonedds` to the repos list under `thirdparty` namespace? https://github.com/osrf/rmf_demos/blob/8e7b734f3961fdb71fa1d896ed0ecc3a96c41870/rmf_demos.repos#L22

This we can we ensure the `0.7.4` tag of `rmf_cyclonedds` gets cloned during setup.

**Comment by marcoag** [comment]
Rviz segfault has been fixed, it seems that `cyclonedds` was segfaulting because a subscription permission was missing. `cyclonedds` should throw a message instead, I believe it does usually. Will further investigate and report this. 

> Also, what do you think of adding `rmw_cyclonedds` and `cyclonedds` to the repos list under `thirdparty` namespace?
> 
> https://github.com/osrf/rmf_demos/blob/8e7b734f3961fdb71fa1d896ed0ecc3a96c41870/rmf_demos.repos#L22
> 
> This we can we ensure the `0.7.4` tag of `rmf_cyclonedds` gets cloned during setup.

`cyclonedds` is only relevant to the SROS 2 demo I would say we leave it out as it's not a mandatory repo? 

Regarding the version of `cyclonedds`, we shouldn't be using 0.7.4 but the latest (0.7.5), I reported the building issue faced: 

https://github.com/ros2/rmw_cyclonedds/issues/261

**Comment by Yadunund** [comment]
Rviz is no longer crashing :+1: 

But after a couple minutes of running the demo I observe this message on all the tmux windows
``` 
fsm: handshake (lguid=e0e9560c:e7a9052e:270b167b:1c1 rguid=d52568ea:4d08201b:d6a90132:1c1) failed: (1) Timed out
```
The simulation becomes unresponsive afterwards. 

>cyclonedds is only relevant to the SROS 2 demo I would say we leave it out as it's not a mandatory repo?

That is a fair point, we can leave it out for now.

**Comment by marcoag** [comment]
Just created a separate issue for the handshake as it's not directly related to the demo itself: https://github.com/osrf/rmf_demos/issues/159

I will follow up on fixes with cyclonedds and update there.

**Review by Yadunund — APPROVED** [review]

**Pull request merged**
